### PR TITLE
Bug fixes - fixes #3559

### LINKED ISSUE #3559: less module resolve algorithm problem
i'm using with bit, project's structure is:

```
boilerplate
├── components/home/sfc-avatar
|  ├── SFCAvatar.less
|  └── node_modules/src/styles/mixins/index.less
└── src/styles/mixins.less
```

and the SFCAvatar.less:

```less
@import 'src/styles/mixins';
```

the less-loader config

```js
{
  test: /\.less$/,
  use: [
    'style-loader',
    'css-loader',
    {
      loader: 'less-loader',
      options: {
        lessOptions: loaderContext => {
          const { context, rootContext } = loaderContext;
          return {
            paths: [
              path.join(context, 'node_modules'),
              path.join(process.cwd(), 'node_modules'),
              process.cwd()
            ]
          };
        }
      }
    }
  ]
}
```

- Operating System: Mac
- Node Version: 10
- NPM Version: 6.4.1
- webpack Version: 4.28.4
- less-loader Version: 7.0.1
- less Version: 3.12.2

### Expected Behavior

SFCAvatar.less should import the `node_modules/src/styles/mixins/index.less`

### Actual Behavior

 `src/styles/mixins.less` imported

https://github.com/webpack-contrib/less-loader/issues/385#issuecomment-699984385
image

here is the demo https://github.com/gogoyqj/less-loader-demo

image

### PR DESCRIPTION
**Pull request by gogoyqj**
try to resolve`src/styles/mixins` like request to specified `paths[i] + src/styles/mixins/index` if `paths[i] + src/styles/mixins` exists(is a directory)

### CODE REVIEW

**Comment by gogoyqj** [comment]
@matthew-dean Could u please have a review？

File: packages/less/src/less-node/file-manager.js [hunk 1]
```diff
@@ -70,13 +70,20 @@ class FileManager extends AbstractFileManager {
                                 fullFilename = path.join(paths[i], fullFilename);
                             }
 
-                            if (!explicit && paths[i] === '.') {
+                            const isNodeModulePath = !!paths[i].match(/[/\\]node_modules[/\\]?/g);
+
+                            if (!explicit && (paths[i] === '.' || isNodeModulePath)) {
```
Comment (aminya) [inline packages/less/src/less-node/file-manager.js:75]: Inline the variable as it does not seem to be used anywhere else
```suggestion
                            if (!explicit && (paths[i] === '.' || /*isNodeModulePath*/ Boolean(paths[i].match(/[/\\]node_modules[/\\]?/g)))) {
```

**Comment by gogoyqj** [comment]
any progress

**Comment by matthew-dean** [comment]
Closing per comment thread here: https://github.com/less/less.js/issues/3559#issuecomment-981392206

**Pull request closed without merging**